src/priority/mod.rs

```diff
@@ -0,0 +1,81 @@
+use bp7::canonical::{new_canonical_block, CanonicalBlockType, CanonicalData};
+use bp7::{CanonicalBlock, EndpointID};
+use derive_try_from_primitive::TryFromPrimitive;
+use serde::de::{SeqAccess, Visitor};
+use serde::ser::{SerializeSeq, Serializer};
+use serde::{de, Deserialize, Deserializer, Serialize};
+use std::convert::TryFrom;
+use std::fmt;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum PriorityError {
+    #[error("serde cbor error: {0}")]
+    Cbor(#[from] serde_cbor::Error),
+    #[error("failed to create endpoint: {0}")]
+    EndpointIdInvalid(#[from] bp7::eid::EndpointIdError),
+    #[error("invalid endpoint supplied")]
+    InvalidEndpoint,
+    #[error("payload missing")]
+    PayloadMissing,
+    #[error("invalid priority block")]
+    InvalidPriorityBlock,
+}
+
+// HOP_COUNT_BLOCK is a BlockType for a Hop Count block as defined in
+// section 4.3.3.
+pub const PRIORITY_BLOCK: CanonicalBlockType = 224;
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct PriorityBlockData(u16);
```

Comment:
a doc string describing the priority system would be nice. is lower better or higher?
it might make sense to have const's or enums for HIGHEST, LOWEST and MEDIUM priorities.
an even more convenient way for the API would be helpers to return a complete block, e.g., `new_highest_priority_block`, `new_lowest_priority_block`, etc.